Marco/feature/onboarding

MarcoFrancisco13 opened the pull request:
- [x] Login Page
- [x] Onboarding Screens

![image](https://user-images.githubusercontent.com/79687819/115967092-384f5280-a563-11eb-8383-202a8cfdbe48.png)

![image](https://user-images.githubusercontent.com/79687819/115967096-3f766080-a563-11eb-8265-9982a4541721.png)

![image](https://user-images.githubusercontent.com/79687819/115967101-44d3ab00-a563-11eb-8693-1dd85fd057d7.png)

devnamme commented on lib/screens/login.dart:
```diff
@@ -0,0 +1,128 @@
+import 'package:flutter/material.dart';
+import 'package:admu_student_app/screens/onboarding.dart';
+
+class login extends StatelessWidget {
```
class names usually start with capital and use camelcase: `class Login extends...` or `class BottomNavBar extends ...`

devnamme commented on lib/screens/onboarding.dart:
```diff
@@ -0,0 +1,111 @@
+  )
+];
+
+class _onboardingState extends State<onboarding> {
```
also here: `class _OnboardingState extends State<Onboarding>`
change the declaration for `class Onboarding` above

dnamme commented:
try using the font from the [google_fonts package](https://pub.dev/packages/google_fonts), as far as i know, `DM Sans` is the font that we're using

dnamme commented:
if possible, also try to fix the spacing for the login page and perhaps convert the onboarding images to plain shapes (it has a padding in the left and right edges since it only uses an image

tho for the latter, i don't know how that package/widget works. in case that it's not possible to remove the padding, you can leave it for now since the onboarding page isn't finalized yet from the design

MarcoFrancisco13 commented:
- [x] Fixed Padding in Login Page
- [x] Changed onboarding screens to solid colors
- [x] Added DM Sans font in all screens

![image](https://user-images.githubusercontent.com/79687819/116114822-4ed4e580-a6ec-11eb-890b-b28e4e06239b.png)

![image](https://user-images.githubusercontent.com/79687819/116114879-5b593e00-a6ec-11eb-9437-59f2a6144ecc.png)

devnamme commented on lib/screens/login.dart:
```diff
@@ -0,0 +1,156 @@
+  Widget build(BuildContext context) {
+    return MediaQuery(
+      data: MediaQueryData(),
+      child: MaterialApp(
```
`lib/main.dart` already uses a `MaterialApp` class to generate the app, having another one (nested) might lead to errors later on (?)

devnamme commented on lib/screens/login.dart:
```diff
@@ -0,0 +1,156 @@
+                      backgroundImage: AssetImage('onboarding.png'),
+                      radius: 100,
+                    ),
+                    Text('Welcome to Beadle!',
```
text isn't centered (visible in smaller screens)
![image](https://user-images.githubusercontent.com/72055285/116119493-14ba1280-a6f1-11eb-9a22-faf0260538fd.png)

devnamme commented on lib/screens/onboarding.dart:
```diff
@@ -0,0 +1,110 @@
+import 'package:flutter/material.dart';
```
although this isn't really a standard, should we stick to a simple format for imports?

```text
dart

flutter and other packages

project
```

example: 
```dart
import 'dart:io';

import 'package:flutter/material.dart';
import 'package:google_fonts/...';

import 'package:admu_student_app/...';
```

MarcoFrancisco13 replied:
got it will make changes in a bit !

dnamme reviewed (approved):
small changes only, the issue with `MaterialApp` can be changed later on when we receive our app icon. you can merge it (after fixing conflicts), let me know if it's done to fix some stuff in develop

MarcoFrancisco13 commented:
- [x] Fixed centering issue on smaller screens
- [x] Arranged imports according to format 

![image](https://user-images.githubusercontent.com/79687819/116127142-11775480-a6fa-11eb-818c-dddd5505c69c.png)

dnamme reviewed (approved):
great job dude! after merging this, the develop branch will still be broken (conflicts, class names, navigation, etc). i'll also let Chris know later in the gc so that we can hopefully have a working dev branch by Tuesday night - which means that creating a new branch before that one will lead to more conflicts. i'll also be sending some changes for the tasks/timeline this week (moving deadlines, etc.)

love those colors in the onboarding screens btw!

dnamme commented:
hello! should i merge this now (and try to fix the conflicts) or will you be the one doing it?
this is the first time seeing that "too complex to resolve in the web editor" message

MarcoFrancisco13 commented:
> 
> 
> hello! should i merge this now (and try to fix the conflicts) or will you be the one doing it?
> this is the first time seeing that "too complex to resolve in the web editor" message

Sorry late reply! I'm not too sure how to resolve the conflicts also, but I'll see if I can finish it all before our meeting tomorrow

dnamme commented:
hmm, i'm guessing that the problem is with `lib/main.dart` since dev has it in the `lib` folder while yours is in `lib/screens` (but that was already fixed before). but thanks! i might also run some tests on my local branch to find out how to fix it
